zio-interop-cats/shared/src/main/scala/zio/interop/cats.scala

```diff
@@ -242,85 +259,125 @@ private class CatsConcurrent[R] extends CatsMonadError[R, Throwable] with Concur
   }
 
   override final def never[A]: RIO[R, A] =
-    ZIO.never
+    ZIO.never(CoreTracer.newTrace)
+
+  override final def async[A](k: (Either[Throwable, A] => Unit) => Unit): RIO[R, A] = {
+    implicit def trace: ZTraceElement = InteropTracer.newTrace(k)
 
-  override final def async[A](k: (Either[Throwable, A] => Unit) => Unit): RIO[R, A] =
     ZIO.async(kk => k(kk apply _.fold(ZIO.fail(_), ZIO.succeedNow)))
+  }
+
+  override final def asyncF[A](k: (Either[Throwable, A] => Unit) => RIO[R, Unit]): RIO[R, A] = {
+    implicit def trace: ZTraceElement = InteropTracer.newTrace(k)
 
-  override final def asyncF[A](k: (Either[Throwable, A] => Unit) => RIO[R, Unit]): RIO[R, A] =
     ZIO.asyncZIO(kk => k(kk apply _.fold(ZIO.fail(_), ZIO.succeedNow)).orDie)
+  }
 
   override final def suspend[A](thunk: => RIO[R, A]): RIO[R, A] =
-    ZIO.suspend(thunk)
+    ZIO.suspend(thunk)(CoreTracer.newTrace)
 
   override final def delay[A](thunk: => A): RIO[R, A] =
-    ZIO.attempt(thunk)
+    ZIO.attempt(thunk)(CoreTracer.newTrace)
```

Comment:
`thunk` is a `Function0` and has a source attached and needs to be traced.

Reply:
fixed